scripts/builder_setup.sh

```diff
@@ -53,9 +68,8 @@ export IBM_MQ_VERSION=9.2.4.0-IBM-MQ-DevToolkit
 # avoiding the error.
 brew untap --force homebrew/cask
 rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
-unset HOMEBREW_NO_INSTALL_FROM_API
 
-CI=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+do_with_retries "CI=1; unset HOMEBREW_NO_INSTALL_FROM_API; $(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" 5
```

Comment:
I am not sure the `CI=1` will be applied to the command here. You should use `export CI=1` or use the syntax `ENV=value command`

Reply:
The `export` makes no difference here, because the `CI=1` command is run in the same shell as the Homebrew install script. Using `ENV=value command` has no effect, unless the next command is a shell invocation. See the following sequence:
```
bash-5.2$ CI=4; echo $CI
4
bash-5.2$ CI=10 echo $CI
4
bash-5.2$ CI=10; echo $CI
10
bash-5.2$ CI=4 echo $CI
10
bash-5.2$ CI=4; echo $CI
4
bash-5.2$ CI=4; /bin/bash -c 'echo $CI'

bash-5.2$ CI=4 /bin/bash -c 'echo $CI'
4
bash-5.2$ CI=10 /bin/bash -c 'echo $CI'
10
bash-5.2$ export CI=4; /bin/bash -c 'echo $CI'
4
```

The action logs show that `$CI` is set: https://github.com/DataDog/datadog-agent-macos-build/actions/runs/8432695067/job/23092268466?pr=181#step:7:14